picard/ui/searchdialog/track.py

```diff
@@ -197,10 +197,11 @@ def load_selection(self, row):
                 # No files associated. Just a normal search.
                 self.tagger.load_album(track["musicbrainz_albumid"])
         else:
-            if self.file_:
+            if self.file_ and getattr(self.file_.parent, 'album', None):
                 album = self.file_.parent.album
-                self.tagger.move_file_to_nat(track["musicbrainz_recordingid"])
+                self.tagger.move_file_to_nat(self.file_, track["musicbrainz_recordingid"])
                 if album._files == 0:
                     self.tagger.remove_album(album)
             else:
                 self.tagger.load_nat(track["musicbrainz_recordingid"], node)
+                self.tagger.move_file_to_nat(self.file_, track["musicbrainz_recordingid"])
```

Comment:
@samj1912 , @zas I'm not sure if I should pass node as third parameter here (and in the same call a few lines above). What do you think?

Reply:
@antlarr, if you don't pass the node, it will re-fetch the NAT from MB. If the node exists and has data, it is better to pass it. It decreases requests.

Reply:
ah, good to know :) I'll pass it then